iOS/DuckDuckGo/TabSwitcherViewController.swift

```diff
@@ -122,38 +126,98 @@ class TabSwitcherViewController: UIViewController {
     }
 
-    fileprivate func createTopBar() {
+    fileprivate func createTitleBar() {
         let appearance = UINavigationBarAppearance()
         appearance.configureWithTransparentBackground()
-        topBarView.standardAppearance = appearance
-        topBarView.scrollEdgeAppearance = appearance
+        titleBarView.standardAppearance = appearance
+        titleBarView.scrollEdgeAppearance = appearance
+    }
+
+    private func activateLayoutConstraintsBasedOnBarPosition() {
+        let isBottomBar = appSettings.currentAddressBarPosition.isBottom
+
+        // Potentially for these 3 we could do thing better for 'normal' on iPad
+        let topOffset = -6.0
+        let bottomOffset = 8.0
+        let navHPadding = isExperimentalThemingEnabled ? 10.0 : 0.0
+
+        // The constants here are to force the ai button to align between the tab switcher and this view
+        NSLayoutConstraint.activate([
+            titleBarView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor, constant: navHPadding),
+            titleBarView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor, constant: -navHPadding),
+            isBottomBar ? titleBarView.bottomAnchor.constraint(equalTo: toolbar.topAnchor, constant: topOffset) : nil,
+            !isBottomBar ? titleBarView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: bottomOffset) : nil,
+
+            collectionView.topAnchor.constraint(equalTo: isBottomBar ? view.safeAreaLayoutGuide.topAnchor : titleBarView.bottomAnchor),
+            collectionView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+            collectionView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+            collectionView.bottomAnchor.constraint(equalTo: isBottomBar ? titleBarView.topAnchor : toolbar.topAnchor),
+
+            // Always at the bottom
+            toolbar.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            toolbar.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            toolbar.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)
+        ].compactMap { $0 })
+    }
+
+    private func setupBarsLayout() {
+        // Remove existing constraints to avoid conflicts
+        titleBarView.translatesAutoresizingMaskIntoConstraints = false
+        toolbar.translatesAutoresizingMaskIntoConstraints = false
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        
+        // Clear existing constraints for these views comprehensively
+        let viewsToRemoveConstraintsFor = [titleBarView, toolbar, collectionView]
+        viewsToRemoveConstraintsFor.forEach { targetView in
+            targetView?.removeFromSuperview()
+        }
+        
+        // Re-add the views to the hierarchy
+        view.addSubview(titleBarView)
+        view.addSubview(toolbar)
+        view.addSubview(collectionView)
```

Comment:
Then maybe we could just deactivate the constraints here without removing and re-adding all subviews. Might be a performance gain. I tried it locally and it seemed to work.